ast/edit/testcase.go

```diff
@@ -0,0 +1,58 @@
+package edit
+
+import (
+	"github.com/influxdata/flux/ast"
+)
+
+/* TestCaseTransform is a macro that transforms ast that uses
+   the testcase statement into multiple independent packages with
+   a single flux script that is the test.
+
+   For instance, the following flux:
+
+import "testing"
+myVar = 4
+testcase addition { test.assertEqual(want: 2 + 2, got: myVar }
+
+	...would be transformed into the equivalent script...
+
+import "testing"
+myVar = 4
+test.assertEqual(want: 2 + 2, got: myVar
+*/
+func TestcaseTransform(pkg *ast.Package) ([]*ast.Package, error) {
+	pkgs := []*ast.Package{}
+
+	var predicate []ast.Statement
+	for _, file := range pkg.Files {
+		for _, item := range file.Body {
+			if _, ok := item.(*ast.TestCaseStatement); ok {
+				continue
+			}
+			predicate = append(predicate, item)
+		}
+	}
+
+	for _, file := range pkg.Files {
+		for _, item := range file.Body {
+			testcase, ok := item.(*ast.TestCaseStatement)
+			if !ok {
+				continue
+			}
+			newPkg := pkg.Copy().(*ast.Package)
+			newPkg.Package = "main"
+			newFile := file.Copy().(*ast.File)
+			newFile.Name = testcase.ID.Name
+			newFile.Package.Name.Name = "main"
+
+			var body []ast.Statement
+			body = append(body, predicate...)
+			body = append(body, testcase.Block.Body...)
+			newFile.Body = body
+			newPkg.Files = []*ast.File{newFile}
+			pkgs = append(pkgs, newPkg)
+		}
+	}
+
+	return pkgs, nil
+}
```

Comment:
So to summarize what this does (I haven't looked at the tests yet), it will take all of the top-level non-testcase statements in each file and put them in a list, and then use them as a prelude to each test case.

Is it possible to have a test package that contains multiple files? I'm not sure myself... I know one of the points of this test effort was to have more than one test case per file, but I'm not sure if it will be possible to have more than one file in a test package?

If it _is_ possible, I'd imagine we might want to do something like this:
```
// file0_test.flux
package foo_test

import "testing"

want = 30
test addition { testing.assertEqual(15 + 15, want) }
```
```
// file1_test.flux
package foo_test

import "testing"

want = 50
test subtraction { testing.assertEqual(100 - 50, want) }
```
But this would result in a file that has
```
...
want = 30
want = 50
testing.assertEqual(15 + 15, want)
```

Should we be taking only the statements from the file in which a test case appears instead?

This isn't relevant unless you can have multi-file test packages, so I guess that's the real question here.

Reply:
I explicitly had a check to make sure that we only ever had a package with one test file in this transformation case, and that was my default. After chatting with @nathanielc about guarding a bunch of things, I stripped that out and just iterated the test files. I had the same concerns, but since we were the only ones consuming this API, I figured it was addressable if we ever came across it. I am _happy_ to add the check back, if you feel strongly about it; I was trying to match convention and idiom here where I couldn't make a _strong_ case for my default.

Reply:
If it's assumed that test packages have just one test file, is that enforced anywhere? I think it'd be good to just have a check here that returns a user error if the number of files in the input package is greater than one.

Reply:
I'm totally happy to do that, as that's what feels right to me.